src/test/java/org/jenkinsci/plugins/envinject/EnvInjectJobPropertyTest.java

```diff
@@ -0,0 +1,134 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.model.Cause;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.ParametersAction;
+import hudson.model.Result;
+import hudson.model.StringParameterValue;
+import hudson.model.TaskListener;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.tasks.Shell;
+import org.junit.Rule;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.*;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.jvnet.hudson.test.CaptureEnvironmentBuilder;
+
+/**
+ * Tests for {@link EnvInjectJobProperty}
+ * @author Oleg Nenashev
+ */
+public class EnvInjectJobPropertyTest {
+    
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testVariableInjection() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, "FOO=BAR", null, null, null, false));
+        prop.setOn(true);
+        project.addProperty(prop);
+        
+        FreeStyleBuild build = j.buildAndAssertSuccess(project);
+        assertEquals("BAR", build.getEnvironment(TaskListener.NULL).get("FOO"));
+    }
+    
+    @Test
+    public void testVariableInjection_PropertyDisabled() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, "FOO=BAR", null, null, null, false));
+        // prop.setOn(false); - It is default
+        project.addProperty(prop);
+        
+        FreeStyleBuild build = j.buildAndAssertSuccess(project);
+        assertNull("The plugin should not inject properties by default", build.getEnvironment(TaskListener.NULL).get("FOO"));
+    }
+    
+    @Test
+    public void testKeepBuildVariables_Default() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setOn(true);
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, null, null, null, null, false));
+        // Default keepBuildVariables should be true
+        project.addProperty(prop);
+        
+        QueueTaskFuture<FreeStyleBuild> scheduled = project.scheduleBuild2(0, new Cause.UserIdCause(),
+                new ParametersAction(new StringParameterValue("PARAM", "Value")));
+        assertNotNull(scheduled);
```

Comment:
please add a little comment for asserts. In standard log its difficult to understand cause of the fail if you have only stack-trace